src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int SHAPE_RANDOM_BOUND = FigureEnum.values().length;
+    private static final int VALUE_RANDOM_BOUND = 20;
+    private static final String DEFAULT_COLOR = Color.WHITE.name().toLowerCase();
+    private static final int DEFAULT_VALUE = 10;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        FigureEnum figure = FigureEnum.values()[random.nextInt(SHAPE_RANDOM_BOUND)];
+        String color = colorSupplier.getRandomColor();
+        int valueFirst = random.nextInt(VALUE_RANDOM_BOUND);
+        int valueSecond = random.nextInt(VALUE_RANDOM_BOUND);
+        int valueThird = random.nextInt(VALUE_RANDOM_BOUND);
```

Comment:
Creating a figure, don't pass expressions in the constructor. Instead, create separate variables for the values and pass them on for better code readability. This means you should assign the random values to variables with meaningful names before using them to create new figure instances.